scripts/publish-npm.js

```diff
@@ -0,0 +1,121 @@
+const { execSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const assert = require('assert');
+const readline = require('readline');
+
+const rootPath = path.resolve('__dirname', '..');
+
+function requireSegmentApiKey() {
+  const MONGOSH_SEGMENT_API_KEY = process.env.MONGOSH_SEGMENT_API_KEY || '';
+
+  if (!MONGOSH_SEGMENT_API_KEY) {
+    throw new Error('MONGOSH_SEGMENT_API_KEY environment variable must be set');
+  }
+
+  if (MONGOSH_SEGMENT_API_KEY.length !== 32) {
+    throw new Error('MONGOSH_SEGMENT_API_KEY does not seem to have the proper format');
+  }
+
+  return MONGOSH_SEGMENT_API_KEY;
+}
+
+function gitClone(repo, dest) {
+  return execSync(`git clone ${repo} ${dest}`);
+}
+
+function getGitRemoteUrl() {
+  return execSync('git config --get remote.origin.url', {cwd: rootPath}).toString().trim();
+}
+
+function getGitRemoteHeadSHA(remoteUrl) {
+  return execSync(
+    `git ls-remote ${remoteUrl} HEAD`, {cwd: rootPath}
+  ).toString().split(/\s+/)[0].trim();
+}
+
+function writeSegmentApiKey(segmentApiKey, releaseDirPath) {
+  const analyticsConfigPath = path.resolve(
+    releaseDirPath,
+    'packages',
+    'cli-repl',
+    'lib',
+    'analytics-config.js'
+  );
+
+  console.info(`writing ${segmentApiKey} to ${analyticsConfigPath}`);
+
+  fs.writeFileSync(
+    analyticsConfigPath,
+    `module.exports = { SEGMENT_API_KEY: '${segmentApiKey}' };`
+  );
+
+  assert.equal(require(analyticsConfigPath).SEGMENT_API_KEY, segmentApiKey);
+}
+
+function isDone(task, releaseDirPath) {
+  return fs.existsSync(`${releaseDirPath}-${task}.done`);
+}
+
+function done(task, releaseDirPath) {
+  fs.writeFileSync(`${releaseDirPath}-${task}.done`, '');
```

Comment:
you're using files to track the state of what was completed, right? So we can restart the npm release process again if something failed?

Will this tmp dir live long enough for us to check this information after an unsuccessful publish?

Reply:
Is a `tmp` dir inside the repo folder (added it to `.gitignore`), will never be removed .. although probably would be nice to delete it after success, or at least do a console log "this folder can be deleted now".

Reply:
Ah gotcha! I think then later on we can also do a clean up for when all tasks have successfully finished and delete the whole dir.